src/SFA.DAS.AssessorService.Application/Handlers/Certificates/GetCertificateAddressesHandler.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using AutoMapper;
+using MediatR;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.AssessorService.Api.Types.Models;
+using SFA.DAS.AssessorService.Application.Interfaces;
+
+namespace SFA.DAS.AssessorService.Application.Handlers.Certificates
+{
+    public class GetCertificateAddressesHandler : IRequestHandler<GetAddressesRequest, List<CertificateAddressResponse>>
+    {
+        private readonly IContactQueryRepository _contactQueryRepository;
+        private readonly ICertificateRepository _certificateRepository;
+        private readonly IIlrRepository _ilrRepository;
+        private readonly IOrganisationQueryRepository _organisationQueryRepository;
+        private readonly ILogger<StartCertificateHandler> _logger;
+
+        public GetCertificateAddressesHandler(
+            IContactQueryRepository contactQueryRepository,
+            ICertificateRepository certificateRepository, IIlrRepository ilrRepository,            
+            IOrganisationQueryRepository organisationQueryRepository, ILogger<StartCertificateHandler> logger)
+        {
+            _contactQueryRepository = contactQueryRepository;
+            _certificateRepository = certificateRepository;
+            _ilrRepository = ilrRepository;        
+            _organisationQueryRepository = organisationQueryRepository;
+            _logger = logger;
+        }
+     
+        public async Task<List<CertificateAddressResponse>> Handle(GetAddressesRequest request, CancellationToken cancellationToken)
+        {
+            var contact = await _contactQueryRepository.GetContact(request.Username);
+            var organisation = await _organisationQueryRepository.Get(contact.EndPointAssessorOrganisationId);
+
+            var certificateAddresses = await _certificateRepository.GetCertificateAddresses(organisation.Id);
```

Comment:
Could this just pass the username to GetCertificateAddresses and then do the link from Contact to Organisation in the sproc?  Save those two extra unneeded db calls? 😄